tests/integration/test_api.py

```diff
@@ -0,0 +1,196 @@
+from __future__ import absolute_import, print_function
+import sys
+import os
+import uuid
+sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", ".."))
+
+from webserver.testing import ServerTestCase
+from flask import url_for
+import db.user
+import time
+import json
+
+# lifted from AcousticBrainz
+def is_valid_uuid(u):
+    try:
+        u = uuid.UUID(u)
+        return True
+    except ValueError:
+        return False
+
+class APITestCase(ServerTestCase):
+
+    def setUp(self):
+        self.user = db.user.get_or_create('testuserpleaseignore')
+
+    def test_get_listens(self):
+        """ Test to make sure that the api sends valid listens on get requests.
+        """
+        with open(os.path.join(os.getcwd(), 'testdata', 'valid_single.json'), 'r') as f:
+            payload = json.load(f)
+        payload['payload'][0]['listened_at'] = int(time.time())
+        response = self.send_data(payload)
+        self.assert200(response)
+        time.sleep(10)
```

Comment:
As this is an integration test between components I don't think there's much else we can do.